crates/tests/test-tedge-cli/src/test_env.rs

```diff
@@ -0,0 +1,19 @@
+pub struct C8YTestEnv {
+    pub url: String,
+    pub user: String,
+    pub pass: String,
+}
+
+impl Default for C8YTestEnv {
+    fn default() -> Self {
+        let url = envvar("C8YURL", "the endpoint to connect c8y");
+        let user = envvar("C8YUSERNAME", "the user to connect c8y");
+        let pass = envvar("C8YPASSWORD", "the password to connect c8y");
+
+        C8YTestEnv { url, user, pass }
+    }
+}
+
+fn envvar(var: &str, doc: &str) -> String {
+    std::env::var(var).unwrap_or_else(|_| format!("The {var} env variable must be set with {doc}"))
```

Comment:
I guess you mean to return a `Result<String>` here and fail with the message (`"The {var} env variable must be set with {doc}"`) if the variable is not there, right?

Reply:
In a test binary I consider it acceptable if it panics on error. No need to push around a Result

Reply:
true. So the `format!()` macro should be replaced with `panic!()` I guess?

Reply:
Oh yeah, duh. I read it as `panic!` already :roll_eyes: My bad!

In that case, yes. I'd suggest to replace format with panic.

Reply:
Oops! I read it as `panic!` too!